File: back/src/main/java/br/eng/eliseu/presente/model/filter/PessoaFilter.java
package br.eng.eliseu.presente.model.filter;

import br.eng.eliseu.presente.model.StatusEnum;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PessoaFilter extends BaseFilter {

    private Long id;

    @NotNull(message = "clienteId é obrigatório")
    private Long clienteId;

    private String nome;
    private String email;
    private String telefone;
    private StatusEnum status;

}

File: front/src/app/shared/model/filter/pessoa-filter.ts
// src/app/shared/model/filter/pessoa-filter.ts
import { BaseFilter } from '../core/base-filter';

export class PessoaFilter extends BaseFilter {

    id?: number;

    clienteId?: number;

    nome?: string;
    cpf?: string;
    email?: string;
    telefone?: string;
    status?: string;

    constructor(init?: Partial<PessoaFilter>) {
        super();
        Object.assign(this, init);
    }

}
